setup.py

```diff
@@ -104,6 +104,10 @@ def main():
                   'rfc959', 'rfc1123', 'rfc2228', 'rfc2428', 'rfc2640',
                   'rfc3659'],
         # fmt: on
+        install_requires=[
+            "pyasyncore;python_version>='3.12'",
+            "pyasynchat;python_version>='3.12'",
+            ],
```

Comment:
I think it's better to remove the 3.12 dependency and always install these 2 libs. Reason: according to https://github.com/simonrob/pyasyncore, they reflect the Python 3.9 incarnation, so I think it's better to stick with that same version also for people using Python < 3.9.

> The [specific version of asyncore that is provided (https://github.com/simonrob/pyasyncore/blob/master/asyncore/__init__.py) is the last update before the addition of deprecation/removal warnings at import time, and is essentially equivalent to [the version bundled with Python 3.9](https://github.com/python/cpython/blob/3.9/Lib/asyncore.py) (which was the last time the module was meaningfully updated).

Reply:
> I think it's better to remove the 3.12 dependency and always install these 2 libs. Reason: according to https://github.com/simonrob/pyasyncore, they reflect the Python 3.9 incarnation, so I think it's better to stick with that same version also for people using Python < 3.9.
> 
> > The [specific version of asyncore that is provided (https://github.com/simonrob/pyasyncore/blob/master/asyncore/**init**.py) is the last update before the addition of deprecation/removal warnings at import time, and is essentially equivalent to [the version bundled with Python 3.9](https://github.com/python/cpython/blob/3.9/Lib/asyncore.py) (which was the last time the module was meaningfully updated).

Fair enough. I can change it, though it won't have any effect on Python < 3.12 as per the [Installation part](https://github.com/simonrob/pyasyncore#installation) of pyasyncore's README:

"The module can be installed for earlier Python versions, but it will have no effect, and the standard library version of asyncore will be used in its place."

Reply:
Oh you're right. Then I think we're fine.